src/azure-cli-core/azure/cli/core/parser.py

```diff
@@ -202,4 +241,5 @@ def _check_value(self, action, value):
                 suggestion_msg += '\n'.join(['\t' + candidate for candidate in candidates])
                 print(suggestion_msg, file=sys.stderr)
 
+            self._make_failure_recovery_recommendations(*self._get_failure_recovery_arguments(action))
```

Comment:
Same here, use information from `self`, like `self.prog` and `self.command_source`, instead of extracting them from `action`. Then append the recommendations to `suggestion_msg` string.

Reply:
`self.command_source`  does not apply to every case. It may be `None` if a subcommand is not under an extension command group, and it could be `None` if an unrecognized argument is passed to an extension command. I'm open to further recommendations on this topic, but for now I just added further comments and made the logic clearer extraction from `action` is necessary.

Reply:
Switched to appending to `suggestion_msg`. Used `self.command_source` and `self.prog` where possible.